### ISSUE TITLE
[Model Request] Add Moondream VLM Architecture

### ISSUE THREAD

**Issue by BharathC0** [keras-team/keras-hub#2549]
Vision Encoder: SigLIP (Sigmoid Loss for Language Image Pre-Training).

Text Decoder: Phi-1.5 (Microsoft's tiny model).

Connector: simple Projection Layer.

 [HuggingFace Official Moondream Card](https://huggingface.co/vikhyatk/moondream2)

**Comment by vamshi567bh** [comment]
@sachinprasadhs can I work on this

**Comment by sachinprasadhs** [comment]
@BharathC0 , Assigned the issue to you, feel free to create a PR once the model is ready, we will be happy to review. Thank you for your interest.

**Comment by vamshi567bh** [comment]
@sachinprasadhs opend pr please review 
https://github.com/keras-team/keras-hub/pull/2553 thanks!
